mm2src/coins/eth.rs

```diff
@@ -1377,7 +1369,7 @@ pub fn signed_eth_tx_from_bytes(bytes: &[u8]) -> Result<SignedEthTx, String> {
 // For ETH it makes even more sense because different ERC20 tokens can be running on same ETH blockchain.
 // So we would need to handle shared locks anyway.
 lazy_static! {
-    static ref NONCE_LOCK: TimedAsyncMutex<()> = TimedAsyncMutex::new(());
+    static ref STATE: Arc<AsyncMutex<()>> = Arc::new(AsyncMutex::new(()));
```

Comment:
`STATE` is too abstract, please choose a more reasonable name for this.
Also, initial problem appeared because `NONCE_LOCK` was shared for *all* coin protocols. It should be the same, e.g. for `ETH/USDT-ERC20`, but different for `BNB/USDT-BEP20`.

Reply:
`STATE` was changed by `NONCE_LOCK`.

Now `NONCE_LOCK` uses `HashMap` wrapped in a `Mutex`, which allows to separate states between different coin protocols.